gem/impero_utils.py

```diff
@@ -223,7 +223,7 @@ def recurse(expr, loop_indices):
 
     @recurse.register(imp.Terminal)
     def recurse_terminal(expr, loop_indices):
-        return temp_refcount(temporaries_set, expr)
+        return temp_refcount(numbering, expr)
```

Comment:
Meh, only `in` is used in `temp_refcount`, fast lookup works just as well on `dict`s as on `set`s.